Documentation/users/ldap-integration.md

```diff
@@ -0,0 +1,125 @@
+
+## Configuring Tectonic Identity for LDAP authentication
+
+Tectonic Identity is configured through the Tectonic Console to allow for LDAP user authentication. The following information is required to integrate with an LDAP server:
+
+* The LDAP server host name and port
+* An LDAP service account capable of querying for users and groups
+* An LDAP base distinguished name (dn), from which the search for users and groups will begin
+* Attributes used to describe users and groups. For example: email and username
+* (Optional) The root CA if certificate CA verification is desired
+
+Use Tectonic Console to enable LDAP authentication in your cluster:
+
+1. Go to *Administration > Cluster Settings*, and click *LDAP*
+
+2. In the LDAP window that opens, enter your host and the (optional) port of the LDAP server in the form `host:port`, and select a verification option:
+
+ * **No SSL:** required if your LDAP host does *not* use SSL.
+ * **Skip verification:** select to skip verification of the CA.
+ * **Root CA:** if selected, enter the PEM data containing the root CAs.
+
+3. Click *Continue* to enter your LDAP Service Account username and password.
+
+question - is this a 'service account' as defined for Tectonic clusters? if so, I want to point readers to creating-service-accounts.md - beth
+
+4. Click *Continue* to enter user search criteria:
+
+  * **Base DN**: The root LDAP directory to begin the user search. Translates to the query: `(&(objectClass=person)(uid=<username>))`.
+  * **Filter**: Filter(s) applied to every user search to limit the results. For example, when a user search executes, object results could be limited to those with an `objectClass` of [person][person-ldap].
+  * **Username Attribute**: The end user's username for login. The field used when searching for users. This is the field users will use to login. For example, a commonName (assuming it's unique) or an email address.
+  question - elsewhere in the doc, RP specified that Tectonic requires that this field be email. is that true? - beth
+  * **User ID Attribute**: String representation of the user's [unique identifier][uid-ldap].
+  * **Email Attribute**: The user's email address.
+  * **Name Attribute**: The user's display name.
+
+5. Click Continue to enter Group information:
+
+  * **Base DN**: The root LDAP directory to begin the group search. Translates to the query: `(&(objectClass=group)(member=<user uid>))`.
+  * **Filter**: (Optional) Filter to apply when searching the directory. For example, when a group search executes, object results could be limited to those with an `objectClass` of [groupOfNames][groupOfNames-ldap].
+  * **User Attribute**: The user field that a group uses to identify that a user is part of a group. For example, groups that specify each member as `member: cn=john,dc=example,dc=org` in the LDAP directory, are using the [Distinguished Name (DN)][dn-ldap] attribute.
+  so - what gets entered in this field? - beth
```

Comment:
It might be good to just use examples.

If the schema looks like:

```
dn: cn=john,dc=people,dc=example,dc=org
objectClass: person
cn: jane
email: jane.doe@example.com

dn: cn=developers,dc=groups,dc=example,dc=org
objectClass: group
cn: developers
member: jane
```

Then we want to match the "cn" of the user with the "member" field of the group. So for this case the config would be

```
userAttr: cn
memberAttr: member
```

If the schema looks like:

```
dn: cn=john,dc=people,dc=example,dc=org
objectClass: person
cn: jane
email: jane.doe@example.com
memberOf: cn=developers,dc=groups,dc=example,dc=org

dn: cn=developers,dc=groups,dc=example,dc=org
objectClass: group
cn: developers
```

Then we want to match the "memberOf" attribute on the user with the "dn" of the group.

```
userAttr: memberOf
memberAttr: dn
```